olive/passes/pytorch/slicegpt.py

```diff
@@ -128,16 +117,20 @@ def _run_for_config(
             100 * (1 - new_embedding_dim / model_adapter.hidden_size),
         )
 
-        train_dataset = get_dataset(config.calibration_data_config.name)["train"]
-        train_loader = prepare_dataloader(
-            dataset=train_dataset,
-            tokenizer=tokenizer,
-            max_seqlen=config.calibration_max_seqlen,
-            batch_size=config.calibration_batch_size,
-            nsamples=config.calibration_nsamples,
-            varied_seqlen=config.varied_seqlen,
-            seed=config.seed,
-        )
+        data_config = validate_config(config.calibration_data_config, DataConfig)
+        dataloader = data_config.to_data_container().create_dataloader(data_root)
+        dataset = [
+            {
+                "input_ids": data[0]["input_ids"].squeeze(),
+                "attention_mask": data[0]["attention_mask"].squeeze(),
+                "labels": data[1].squeeze(),
+            }
+            for data in dataloader
+        ]
+
+        torch.manual_seed(config.seed)
+        sampler = SubsetRandomSampler(torch.randperm(len(dataset))[: config.calibration_nsamples])
```

Comment:
This looks okay for now. But we can probably add an option in the dataloader section for random sampling so that this extra work of rewrapping the data with a dataloader is not needed. 

Will also remove the confusion between `batch_size`, `max_samples` in the data config and `calibration_batch_size`, `calibration_nsamples` here.